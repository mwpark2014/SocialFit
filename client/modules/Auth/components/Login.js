import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router';
import { FormGroup, FormControl, Button,
  ControlLabel, Row, Col } from 'react-bootstrap';

export default class Login extends Component {
  constructor(props) {
    super(props);

    this.state = { name: '', username: '', email: '', password: '', password2: '' };
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleInputChange = this.handleInputChange.bind(this);
  }

  handleInputChange(event) {
    const target = event.target;
    const value = target.value;
    const name = target.name;
  
    this.setState({
      [name]: value,
    });
  }

  handleSubmit(event) {
    event.preventDefault();
    this.props.loginUser({ username: this.state.username,
                   password: this.state.password });
    // Clear all the fields
    this.state = { name: '', username: '', email: '', password: '', password2: '' };
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <FormGroup>
          <Row>
            <Col md={2}>
              <ControlLabel>Username</ControlLabel>
            </Col>
            <Col md={4}>
            <FormControl type="text" name="username" placeholder="Username" // eslint-disable-line
              onChange={this.handleInputChange} value={this.state.username} required />
            </Col>
          </Row>
        </FormGroup>
        <FormGroup>
          <Row>
            <Col md={2}>
              <ControlLabel>Password</ControlLabel>
            </Col>
            <Col md={4}>
            <FormControl type="password" name="password" placeholder="Password" // eslint-disable-line
              onChange={this.handleInputChange} value={this.state.password} required />
            </Col>
          </Row>
        </FormGroup>
        <Button type="submit">Login</Button>
        <Link to="/Register">Sign Up</Link>
      </form>
    );
  }
}

Login.propTypes = {
  errorMessage: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
  loginUser: PropTypes.func.isRequired,
};
