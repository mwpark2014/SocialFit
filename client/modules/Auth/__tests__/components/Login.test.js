import React from 'react';
import Login from '../../components/Login';
import renderer from 'react-test-renderer';

const loginUser = jest.fn(); //eslint-disable-line

describe('Test render & styles', () => {
  test('Login component snapshot', () => {
    const component = renderer.create(
      <Login
        loginUser={loginUser} errorMessage=""
        message=""
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot(); // eslint-disable-line
  });
});
