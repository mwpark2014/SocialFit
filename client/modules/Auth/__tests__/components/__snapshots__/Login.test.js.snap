// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test render & styles Login component snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <div
      className="row"
    >
      <div
        className="col col-md-4"
      >
        <label
          className=""
          htmlFor={undefined}
        >
          Username
        </label>
      </div>
      <div
        className="col col-md-7"
      >
        <input
          className="form-control"
          name="username"
          onChange={[Function]}
          placeholder="Username"
          required={true}
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="form-group"
  >
    <div
      className="row"
    >
      <div
        className="col col-md-4"
      >
        <label
          className=""
          htmlFor={undefined}
        >
          Password
        </label>
      </div>
      <div
        className="col col-md-7"
      >
        <input
          className="form-control"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col col-md-7 offset-md-4"
    >
      <button
        className="btn btn-secondary btn-block"
        onClick={[Function]}
        type="submit"
      >
        Login
      </button>
    </div>
  </div>
</form>
`;
